ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/manager/bounded/BoundedQueueTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.manager.bounded;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+
+import org.junit.Test;
+
+public class BoundedQueueTest {
+
+  @Test
+  public void offerShouldAcceptNewElements() {
+    int size = 10;
+    final BoundedQueue queue = new BoundedQueue(size, "test", new NoOpMetricsSystem());
+    for (int i = 0; i < size; i++) {
+      assertTrue(queue.offer(() -> {}));
+      assertEquals(i + 1, queue.size());
+    }
+  }
+
+  @Test
+  public void offerShouldMakeARoomAndAcceptNewElementAtFullCapacity() {
+    final BoundedQueue queue = new BoundedQueue(1, "test", new NoOpMetricsSystem());
```

Comment:
Can we update this to have a queue size of 2 or 3 so that we can actually assert that the oldest entry was removed when it filled up?  The default instance equality of `Runnable` will be enough to check which are still in the queue if you store them in variables:
```
Runnable task1 = () -> {};
Runnable task2 = () -> {};
...
assertThat(queue).contains(task1);
assertThat(queue).doesNotContain(task2);
```

And one more little nit - we try to use AssertJ's `Assertion` instead of the Unit `assertTrue`/`assertEquals` as it reads better and gives much better failure messages.

Reply:
Ok. i do it and i update to use AssertJ.